bare/main.go

```diff
@@ -46,11 +49,23 @@ func main() {
 		{config: "bare/config/runner.jsonnet", binary: "bb_runner"},
 		{config: "bare/config/browser.jsonnet", binary: "bb_browser"},
 	}
+	var commands []*exec.Cmd
 	for _, bb := range bbs {
 		log.Println("Starting", bb.binary)
-		go bbStart(bb)
+		commands = append(commands, bbStart(bb))
 	}
 
+	// Kill all started processes if interrupted:
+	c := make(chan os.Signal)
+	signal.Notify(c, os.Interrupt, syscall.SIGTERM)
+	go func() {
+		<-c
+		for _, command := range commands {
+			log.Printf("Killing proc: %d", command.Process.Pid)
+			command.Process.Kill()
+		}
+		os.Exit(1)
+	}()
 	// Wait forever or until stopped
 	<-(chan int)(nil)
```

Comment:
Don't know if I'm right, but why have this forever channel wait and putting the signal handler in a go routine? Can't the signal handler be extracted to the main function level?